bip-v1-packages.mediawiki

```diff
@@ -0,0 +1,346 @@
+|-
+|blockhash || uint256 || 32 || The chain tip at which this package is defined.
+|-
+|txns_length||CompactSize||1 or 3 bytes|| The number of transactions provided.
+|-
+|txns||List of wtxids||txns_length * 32|| The wtxids of each transaction in the package.
+|}
+
+
+# The "pkginfo1" message has the structure defined above, with pchCommand == "pkginfo1".
+
+# A "pkginfo1" message contains information about a version 1 package (defined below), referenced by the wtxid of the transaction it pertains to and the current blockhash.
+
+# Upon receipt of a "pkginfo1" message, the node should decide if it wants to validate the package, request transaction data if necessary, etc. If the 'blockhash' field does not match the node's current best chain tip, the message should be discarded.
+
+# Upon receipt of a malformed "pkginfo1" message or package that does not abide by the rules specified, the sender may be disconnected.
+
+# A node MUST NOT send a "pkginfo1" message that has not been requested by the recipient using "getdata(MSG_PKG1)". Upon receipt of an unsolicited "pkginfo1", a node may disconnect the sender.
+
+# A "pkginfo1" message should only be sent if both peers agreed to send version 1 packages in the version handshake. If a "pkginfo1" message is received from a peer with which package relay was not negotiated, the sender may be disconnected.
+
+====MSG_PKG1====
+
+# A new inv type (MSG_PKG1 == 0x6) is added, for use in inv messages and getdata requests pertaining to version 1 packages.
+
+# As an inv type, it indicates that both transaction data and version 1 package information are available for the transaction. The transaction is referenced by its wtxid. As a getdata request type, it indicates that the sender wants package information for the transaction.
+
+# Nodes may request package information using "getdata(MSG_PKG1)" even if it was not announced using "inv(MSG_PKG1)".
+
+# Nodes should not request package information from peers that have a different best chain tip, since they cannot verify that the package is child-with-unconfirmed-parents unless the 'blockhash' field matches their chain tip.
+
+# Upon receipt of a "getdata" request for "MSG_PKG1", the node should respond with the version 1 package corresponding to the requested transaction and its current chain tip, or with NOTFOUND. The node should not assume that the sender is requesting the transaction data as well.
+
+===Child With Unconfirmed Parents Packages===
+
+Version 1 package relay supports child-with-unconfirmed-parents packages. All v1 packages
+sent between nodes MUST abide by the rules below, otherwise the package is malformed and the sender
+should be disconnected.
+
+A version 1 or '''child-with-unconfirmed-parents''' package can be defined for any transaction that spends
+unconfirmed inputs. The child can be thought of as the "representative" of the package. This package
+can be uniquely identified by the transaction's wtxid and the current chain tip block hash.
```

Comment:
The child's wtxid does not commit to the wtxids of its ancestors (just the txids), so I don't think it's correct to say that a package is uniquely identified by the child transaction's wtxid and the current chain tip.  (I tried to bring this point up in the mailing list thread as well, in the context of package de-duplication.)

Reply:
Yes sorry, I think I should have said the package can be *easily reconstructed* using the transaction's wtxid and current chain tip block hash.